Tests/SentryTests/Transaction/SentrySpanTests.swift

```diff
@@ -451,4 +468,79 @@ class SentrySpanTests: XCTestCase {
         XCTAssertEqual(nameForSentrySpanStatus(.outOfRange), kSentrySpanStatusNameOutOfRange)
         XCTAssertEqual(nameForSentrySpanStatus(.dataLoss), kSentrySpanStatusNameDataLoss)
     }
+    
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+    func testAddSlowFrozenFramesToData() {
+        let (displayLinkWrapper, framesTracker) = givenFramesTracker()
+        
+        let sut = SentrySpan(context: SpanContext(operation: "TEST"), framesTracker: framesTracker)
+        
+        let slow = 2
+        let frozen = 1
+        let normal = 100
+        displayLinkWrapper.givenFrames(slow, frozen, normal)
+        
+        sut.finish()
+        
+        expect(sut.data["frames.total"] as? NSNumber) == NSNumber(value: slow + frozen + normal)
+        expect(sut.data["frames.slow"] as? NSNumber) == NSNumber(value: slow)
+        expect(sut.data["frames.frozen"] as? NSNumber) == NSNumber(value: frozen)
+    }
+    
+    func testDontAddAllZeroSlowFrozenFramesToData() {
+        let (_, framesTracker) = givenFramesTracker()
+        
+        let sut = SentrySpan(context: SpanContext(operation: "TEST"), framesTracker: framesTracker)
+        
+        sut.finish()
+        
+        expect(sut.data["frames.total"]) == nil
+        expect(sut.data["frames.slow"]) == nil
+        expect(sut.data["frames.frozen"]) == nil
```

Comment:
If we're sticking to nimble, please let's keep using @philipphofmann approach, the verbose one is to awful to read.

Reply:
Obviously I disagree that it's awful to read, or I wouldn't have suggested the changes at all

Reply:
I started with `.to(beNil())` but swiftlint fixed it to `== nil`. At first, I thought it's awful to read, but then I kind of started to like it. It's the same syntax you use in if statements, so you are already familiar with it. `.to(beNil())` is a new concept. I'm lazy, so I try to avoid learning new concepts 😄 
I have no strong opining here, but if we go for `.to(beNil())` we also need to change the swiftlint rules.

Reply:
haha, interesting. i'm surprised swiftlint even knows about nimble functions, i guess this means this is a more widespread preference. i'm fine to leave them as they are